src/providers/gce/mod.rs

```diff
@@ -62,6 +62,10 @@ impl GceProvider {
         }
 
         keys.append(&mut self.fetch_ssh_keys("project/attributes/sshKeys")?);
+        if !keys.is_empty() {
```

Comment:
This re-uses `keys` from the instance-level metadata, so it may result in skipping new-endpoint project keys because of instance ones (which I think is unexpected behavior).

I suggest to split and rename into two variables, `instance_keys` and `project_keys`, and also add a comment at the top detailing the deprecated endpoint and referencing GCloud docs.

Reply:
Good catch. Actually, what do you prefer to do if both new-endpoint and old-endpoint exist for project level? Return both? Or check new and fall back to old?